flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "bootloader-experimentation";
+
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+
+  outputs =
+    { self
+    , nixpkgs
+    , ...
+    } @ inputs:
+    let
+      nameValuePair = name: value: { inherit name value; };
+      genAttrs = names: f: builtins.listToAttrs (map (n: nameValuePair n (f n)) names);
+      allSystems = [ "x86_64-linux" "aarch64-linux" "i686-linux" "x86_64-darwin" ];
+
+      forAllSystems = f: genAttrs allSystems (system: f {
+        inherit system;
+        pkgs = import nixpkgs { inherit system; };
```

Comment:
I prefer `import`ing Nixpkgs because it allows for extension down the line (e.g. need to overlay a package for our purposes, or need to specify some Nixpkgs config).